examples/html/supercells.html

```diff
@@ -0,0 +1,557 @@
+	 * */
+	constructor (conf, kind, id, C) {
+		super(conf, kind, id, C)
+
+		/** the variable that is checked against subcells
+		 * @type{Number}
+		 */
+		this.host = this.id
+
+		/** holds the Supercell's @class{SubCell} SubCell objects 
+		 * @type{Array}
+		*/
+		this.subcells = []
+
+		/** encode cell-specific target volume for a simple growth algorithm such that the V can be halved after division
+		 * @type{Number}
+		*/
+		this.V = this.conf["V"][kind]
+	}
+
+
+	/**
+	 *  Halves target volume as post-division volume pressure can lead to artefacts
+	 * (with supercells specifically, the supercell can kill subcells through volume pressure if this is left out)
+	 * @param {Cell} parent - the parent (or other daughter) cell
+	 */ 
+	birth(parent){
+		super.birth(parent)
+		this.V /= 2
+		parent.V /=2
+	}
+
+	/**
+	 * Private function for SubCell, watch out when calling this. 
+	 * Adds a SubCell to the subcells array. 
+	 * @param {Cell} cell - the subcell to track from the supercell
+	 */ 
+	addSubCell(cell){
+		if (!this.subcells.includes(cell)){
+			this.subcells.push(cell)
+		} else {
+			throw("Tried to add subcell when already in subcells", this, cell)
+		}
+	}
+
+	/**
+	 * Private function for SubCell, watch out when calling this. 
+	 * Removes a SubCell to the subcells array. 
+	 * @param {Cell} cell - the subcell to stop tracking from the supercell
+	 */ 
+	removeSubCell(cell){
+		let ix =  this.subcells.indexOf(cell)
+		if (ix !==-1){
+			this.subcells.splice(ix, 1)
+		} else {
+			throw("Tried to remove subcell that was not tracked by this host", this, cell)
+		}
+	}
+
+	/**
+	 * Computes torus-enabled centroid for the pixels handed to it
+	 * Direct copy from Stat CentroidsWithTorusCorrection, but expects pixels instead of cellId
+	 * @param {Array} pixels for host and subcells
+	 * @return {ArrayCoordinate}  centroid of complete host+subcells space
+	 */ 
+	computeHostCentroid(pixels){
+		// copied Torus corrected centroid, but now you can hand all pixels of host+subcells
+		let halfsize = new Array( this.C.ndim).fill(0)
+		for( let i = 0 ; i < this.C.ndim ; i ++ ){
+			halfsize[i] = this.C.extents[i]/2
+		}
+		let cvec = new Array(this.C.ndim).fill(0)
+		for( let dim = 0 ; dim <this.C.ndim ; dim ++ ){
+			let mi = 0.
+			const hsi = halfsize[dim], si = this.C.extents[dim]
+			for( let j = 0 ; j < pixels.length ; j ++ ){
+				let dx = pixels[j][dim] - mi
+				if( this.C.grid.torus[dim] && j > 0 ){
+					if( dx > hsi ){
+						dx -= si
+					} else if( dx < -hsi ){
+						dx += si
+					}
+				}
+				mi += dx/(j+1)
+			}
+			if( mi < 0 ){
+				mi += si
+			} else if( mi > si ){
+				mi -= si
+			}
+			cvec[dim] = mi
+		}
+		return cvec
+	}
+
+	/**
+	 * Let this cell compute a dividing line perpendicular to the longest axis
+	 * Divides the host and all subcells on this line
+	 * Sets new host id for all subcells in new daughter
+	 * 
+	 * @return {Number}  New host CellId
+	 */ 
+	divideHostCell(){
```

Comment:
I've tried a few algorithms for the inheritance of subcells to daughter hosts, and eventually came to the conclusion that the most neat way to do it was to reimplement divideCell(). 
My initial attempts were to divideCell() the host and then try to retrieve through things such as neighbors or centroids which daughter the subcells belong to. This can easily give problems when a host only has a single large subcell, and sometimes give weird encapsulation artefacts as some subcells got completely unlinked from the supercell. 
The best way i found was to divide all subcells along the dividing line of the supercell, which requires the centroid of the division to be calculated on the whole cell (super + subs) and all subcells to get the position of the dividing line. I thought the easiest way to do this was to reimplement the whole function (another way may be to at least return the full dividing line from divideCell(), but I thought this usecase was maybe still too niche to do this).

Reply:
Okay, then let's leave it as is for now, and reevaluate if we find later that more examples need to reimplement divideCell.